dashboard/app/controllers/projects_controller.rb

```diff
@@ -571,6 +571,43 @@ def limited_gallery?
     !project_validator && limited_project_gallery
   end
 
+  # GET /projects/:channel_id/extra_links
+  # Get the extra links for the project for use by project validators.
+  # This is used by lab2 levels that cannot use the haml 'extra links' box since
+  # this box will not refresh when changing levels.
+  def extra_links
+    return render(status: :forbidden, json: {error: 'This user does not have project validator access.'}) unless current_user.project_validator?
+    src_channel_id = params[:channel_id]
+    if src_channel_id == "undefined"
+      return render json: {error: 'No channel id provided.'}, status: :ok
```

Comment:
Should the status be `ok` in this situation?

Reply:
I added this because I saw an error response for `/extra_links` when a project validator was on a non-standalone project level and I don't necessarily think of this as an error. I think I meant for this to be an `ok` response with a different response message...maybe something like:
`render json: {message: 'Levels that are not standalone projects do not include project info.'}, status: :ok`

What do you think?

Reply:
By "non-standalone project level" do you mean any level that's not the "new <lab name> project" level or a level that doesn't have a project at all? We should still be able to get project validator information for a project-backed level, which isn't necessarily the standalone project level.

Reply:
I've noticed that for channel-backed legacy labs like Sprite lab, I don't think we display project/owner info on a non-standalone project level - this ''s/allthethings/lessons/36/levels/2' and level name is 'AllTheThings sprite lab with video'.
![Screenshot 2024-06-21 at 5 33 09 PM](https://github.com/code-dot-org/code-dot-org/assets/107423305/328f9d09-0c7c-4ed9-af9b-cf7a54fb973e)

But once we get the share URL via `dashboard.project.getShareUrl()`, we can then get the project info.
![Screenshot 2024-06-21 at 5 33 30 PM](https://github.com/code-dot-org/code-dot-org/assets/107423305/8474bb4c-9159-4c6d-a476-283cc968fddf)
The level name for the level via share URL is 'New Sprite Lab Project'.

I remember this debugging ZenDesk issues. Makes sense according to: https://github.com/code-dot-org/code-dot-org/blob/6a67ac98118222f97c0208ae60829d207e2cf06d/dashboard/app/views/levels/_admin.html.haml#L115

But for project validators, we will display report abuse and project-sharing info (will be included in follow-up).

Reply:
Ok makes sense. As a follow-up we might be able to improve this to show the project info even for non-standalone project levels, since we know the channel id.

Reply:
But this is definitely not ideal considering that `aichat` doesn't have a standalone project level yet. Can look into how `channel_id` is assigned in redux in a follow-up and find a workaround for this. Currently for non-standalone project levels, `channel_id` is assigned`'undefined'`.

Reply:
Thanks! Think our lines got crossed. Will definitely log a ticket to look into this.

Reply:
Hey Molly - I totally forgot I intentionally did not display project/owner info since this is not shown in legacy labs. I won't show all irrelevant project info (like featured project info or remix ancestry), but will show project id, S3 sources link, owner name and storage_id. Thanks!

Reply:
Here's an updated screenshot of channel-backed level:
![Screenshot 2024-06-24 at 1 33 51 PM](https://github.com/code-dot-org/code-dot-org/assets/107423305/2a2ada25-8075-45bf-8d7e-1e7283d1cc76)